### DISCUSSION TITLE
Removal of support for string[] in 7.0.0, how to proceed?

### DISCUSSION THREAD

**Discussion by DelliriumX** [pillarjs/path-to-regexp#310] [Q&A] [answered]
How does one proceed now that this option is gone?

I was using the feature and I've not managed to find any guidelines on how to transition from having `string[]` to just `string`

**Comment by mxxk** [comment]
@DelliriumX I was also relying on the `string[]` support, so with `path-to-regexp` v7, I ended up implementing a helper similar to what `path-to-regexp` dropped in https://github.com/pillarjs/path-to-regexp/commit/db3beffd99c176c7b16b5b55d7fa0c1c09dec365#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80L624-L634:

https://github.com/pillarjs/path-to-regexp/blob/75e08fcda89cd90f070cd189f3bea0d0e83d5e9c/src/index.ts#L624-L634

<details><summary>For <code>pathToRegexp</code> v7, here's the helper I ended up with.</summary>
<br />

```js
import { pathToRegexp } from 'path-to-regexp';

/**
 * Converts an iterable of paths into a regular expression.
 *
 * This helper is necessary since `path-to-regexp` dropped support for arrays
 * in 7.0.0.
 *
 * @see https://github.com/pillarjs/path-to-regexp/commit/db3beffd99c176c7b16b5b55d7fa0c1c09dec365
 */
function pathsToRegexp(paths) {
  const parts = Array.from(
    paths,
    (path) =>
      pathToRegexp(path, {
        loose: false,
        strict: true,
      }).source,
  );
  return new RegExp(`(?:${parts.join('|')})`, 'i');
}
```
</details>

However, `path-to-regexp` v8 removed the direct ability to get at the regexp, except using an undocumented `re` property.

<details><summary>For <code>pathToRegexp</code> v8, here's the helper I ended up with.</summary>
<br />

```js
import * as pathToRegexp from 'path-to-regexp';

/**
 * Converts an iterable of paths into a regular expression.
 *
 * This helper is necessary since `path-to-regexp` dropped support for arrays
 * in 7.0.0.
 *
 * @see https://github.com/pillarjs/path-to-regexp/commit/db3beffd99c176c7b16b5b55d7fa0c1c09dec365
 */
function pathsToRegexp(paths) {
  const parts = Array.from(paths, (path) => pathToRegexp.match(path).re.source);
  return new RegExp(`(?:${parts.join('|')})`, 'i');
}
```
</details>

The only reassurance regarding the `re` property is that it's used in a `path-to-regexp` redos script:

https://github.com/pillarjs/path-to-regexp/blob/a43e545555bbbb49d08cb7dc1296e0a739c05ea7/scripts/redos.ts#L11

**Reply by blakeembrey** [reply]
Heads up that v8 already did support arrays: https://github.com/pillarjs/path-to-regexp/blob/ed1095e0fa78a692e7f3d489e383e7bb1f9d2cc4/src/index.ts#L507 (I've updated the release notes to correct this).

Only v7 was missing the support, working on adding it back now 😄 Sorry for that, I don't think it's a heavily used feature so it was removed while I was trimming the package size in v7.

**Reply by mxxk** [reply]
Oh snap, I totally missed the fact that `match` supported arrays of paths in v8. Adopting this `match` signature in my case seems like the cleaner approach compared to relying on the regular expression. Thanks for the heads up!

**Comment by blakeembrey** [comment]
Thanks for the question. I tried to clean up features and it's fine to add them back if there are users of them. I'll add back array support in v7, which I already did for v8.

**Reply by DelliriumX** [reply]
This would be useful, thank you. I can implement my workaround as @mxxk suggested while I wait, but would be good to have this out of the box

**Reply by DelliriumX** [reply]
Question, you say "which I already did for v8", i didnt see it mentioned in the release notes. Is it something coming in a future update?
When can we expect this to arrive, for either v7 or v8?

**Reply by blakeembrey** [reply]
Good call, it's just in the README as "String or array of strings.". I'll add it to the release notes too.

**Reply by blakeembrey** [reply, marked as answer]
Added in v7: https://github.com/pillarjs/path-to-regexp/releases/tag/v7.2.0

Arrays were supported already in v8.
